src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,30 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+
+type Props = {
+  todo: Todo | null;
+  onClose: () => void;
+};
+
+export const TodoModal: React.FC<Props> = ({ todo, onClose }) => {
+  const [user, setUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    if (!!todo) {
+      getUser(todo?.userId)
+        .then(res => setUser(res))
+        .catch(() => setUser(null));
+    }
+  }, []);
```

Comment:
```suggestion
  }, [todo]);
```

Reply:
I intend to run this only once, also if I add todos as a dependency useEffect in the render loop infinitely.